lib/puppet/type/netapp_qtree.rb

```diff
@@ -8,23 +8,54 @@
   ensurable
 
   newparam(:name) do
-    desc "The qtree name."
     isnamevar
+    desc "The qtree name."
     validate do |value|
-      unless value =~ /^[\w\-]+$/
+      unless (value =~ /^[\w\-\.]+$/) or (value =~ /^\/\w+\/[\w\-\.]+$/)
          raise ArgumentError, "%s is not a valid qtree name." % value
       end
     end
   end
 
   newparam(:volume) do
     desc "The volume to create qtree against."
+    defaultto do
+      if @resource[:name].count('/') == 2
+        @resource[:name].split('/')[1] 
+      else
+        ""
+      end
+    end
```

Comment:
It seems there's still some problem as the provider generate a list of qtree to match for existance and it use just the qtreename as key so if 2 exists the second overwrite the first and the provider will try to recreate the qtree that already exist on the first volume.

The qtree test2 exist on voltst11 and on voltst12 (created correctly by the first run but the second run try to create test2 on voltst11 but also if it's already created.
```
    netapp_qtree { '/voltst11/test2':
      ensure => present,
      name   => 'test2',
      volume => 'voltst11',
    }
    netapp_qtree { '/voltst12/test2':
      ensure => present,
      name   => 'test2',
      volume => 'voltst12',
    }

```

Relevant logs:
```
Debug: Puppet::Provider::Netapp_qtree.cmode.prefetch: Checking if this is an actual qtree, not a volume. 
Debug: Puppet::Provider::Netapp_qtree.cmode.prefetch: Processing rule for qtree 'test2'.
Debug: Puppet::Provider::Netapp_qtree.cmode.prefetch: Volume for 'test2' is 'voltst11'.
Debug: Creating instance for 'test2'. 

....
Debug: Puppet::Provider::Netapp_qtree.cmode.prefetch: Checking if this is an actual qtree, not a volume. 
Debug: Puppet::Provider::Netapp_qtree.cmode.prefetch: Processing rule for qtree 'test2'.
Debug: Puppet::Provider::Netapp_qtree.cmode.prefetch: Volume for 'test2' is 'voltst12'.
Debug: Creating instance for 'test2'. 

Debug: Puppet::Provider::Netapp_qtree.cmode: checking existance of Netapp qtree test2 against volume voltst11
Debug: Value = 
Debug: Puppet::Provider::Netapp_qtree.cmode: creating Netapp Qtree test2 on volume voltst11.
Debug: apicommand is a - String
Debug: Args array length = 4
Debug: Executing api call qtree-create qtree test2 volume voltst11
Debug: Facter: custom fact "url" was not found.
Error: Executing api call qtree-create qtree test2 volume voltst11 failed: "Failed to create qtree \"test2\" in volume \"voltst11\" Vserver \"ITNAPSVMTEST01\". Reason: Qtree (or directory) \"test2\" already exists."
Error: /Stage[main]/Profile::Storage::Netapp/Netapp_qtree[/voltst11/test2]/ensure: change from absent to present failed: Executing api call qtree-create qtree test2 volume voltst11 failed: "Failed to create qtree \"test2\" in volume \"voltst11\" Vserver \"ITNAPSVMTEST01\". Reason: Qtree (or directory) \"test2\" already exists."
Debug: Puppet::Provider::Netapp_qtree.cmode: checking existance of Netapp qtree test2 against volume voltst12
Debug: Value = present

```
Tomorrow I'll try to check if changing the provider is something I can do.